x-pack/plugins/security_solution/public/actions/toggle_column/cell_action/toggle_column.ts

```diff
@@ -66,6 +74,35 @@ export const createToggleColumnCellActionFactory = createCellActionFactory(
 
       const defaults = isTimelineScope(scopeId) ? timelineDefaults : tableDefaults;
       const { columns } = selector(store.getState(), scopeId) ?? defaults;
+      const alertTableConfigurationId = getAlertConfigIdByScopeId(scopeId);
+      if (alertTableConfigurationId) {
+        const alertsTableConfigurationRegistry =
+          services.triggersActionsUi.alertsTableConfigurationRegistry.get(
+            alertTableConfigurationId
+          );
+        alertsTableConfigurationRegistry?.actions?.toggleColumn(field.name);
+      } else {
```

Comment:
A couple of things here:

- Would it be okay to move this block to the top, after the `if (!scopeId) return;` and before the `getScopedActions(scopeId)`? So we do the special case first, something like:

```
      if (!scopeId) return;

      // Check if the scopeId is a triggersActionsUi AlertsTable
      const alertTableConfigurationId = getAlertConfigIdByScopeId(scopeId);
      if (alertTableConfigurationId) {
        const alertsTableConfigurationRegistry =
          services.triggersActionsUi.alertsTableConfigurationRegistry.get(
            alertTableConfigurationId
          );
        alertsTableConfigurationRegistry?.actions?.toggleColumn(field.name);
        return;
      }

      // For the rest of scopeIds dispatch a regular redux action
      const scopedActions = getScopedActions(scopeId);
```

- The default case with the regular redux actions dispatches below seems duplicated, can we clean it? Also, the currect code is always dispatching them, even after calling the `alertsTableConfigurationRegistry?.actions?.toggleColumn(field.name);`. Is this correct? (the example code above assumes they should not be called so it just `return`)